app/assets/js/app/modules/inline-feedback.js

```diff
@@ -0,0 +1,124 @@
+import domready from './domready'
+
+export const classCallToAction = 'feedback__action'
+export const classInlineForm = 'feedback__inline'
+export const classExpanded = 'is-expanded'
+export const classHidden = 'u-hidden'
+export const classFeedbackThanks = 'feedback__thanks'
+export const classFeedbackCancel = 'feedback__cancel'
+
+export const classMessage = 'js-feedback-message'
+export const className = 'js-feedback-name'
+export const classEmail = 'js-feedback-email'
+export const classSubmit = 'js-feedback-submit'
+
+export const attrHidden = 'aria-hidden'
+
+export const feedbackEndPoint = '/feedback'
+
+class Feedback {
+  constructor(form, callToAction, thanks, cancel) {
+    // This module relies on browsers having implemented
+    // the FormData Interface
+    // (https://developer.mozilla.org/en-US/docs/Web/API/FormData)
+    // < IE10 don't have it so they get the No-JS fallback
+    if (window.FormData) {
+      this.form = form
+      this.registerForm(this.form)
+
+      this.callToAction = callToAction
+      this.registerCallToAction(this.callToAction)
+
+      this.thanks = thanks
+
+      this.cancel = cancel
+      this.registerCancel(this.cancel)
+    }
+  }
+
+  registerForm(element) {
+    element.setAttribute(attrHidden, 'true')
+
+    let elSubmit = element.getElementsByClassName(classSubmit)[0]
+    this.registerSubmit(elSubmit)
+  }
+
+  registerSubmit(element) {
+    element.addEventListener('click', e => {
+      e.preventDefault()
+
+      this.sendFeedback()
+      this.collapse(this.form)
+      this.hide(this.callToAction)
+      this.show(this.thanks)
+    })
+  }
+
+  registerCallToAction(element) {
+    element.addEventListener('click', e => {
+      e.preventDefault()
+
+      this.toggle(this.form)
+    })
+  }
+
+  registerCancel(element) {
+    element.addEventListener('click', e => {
+      e.preventDefault()
+
+      this.collapse(this.form)
+    })
+  }
+
+  sendFeedback() {
+    const form = this.form.getElementsByTagName('form')[0]
+    const formData = new FormData(form)
+    formData.set('redirect', 'false')
```

Comment:
The advanced interface for FormData is not implemented across all browsers. Need to polyfill or handle redirect in another way other than manipulating the form.